scripts/merkl_incentive_campaign.py

```diff
@@ -0,0 +1,325 @@
+            "address[]",  # blacklist
+            "string",  # reason
+        ],
+        [
+            r.uniswap.v3pool_wbtc_weth,  # @note TBD! using a placeholder for script testing
+            weight_fees,  # propFees
+            weight_token_a,  # propToken0
+            weight_token_b,  # propToken1
+            0,  # isOutOfRangeIncentivized
+            AddressZero,  # boostingAddress
+            0,  # boostedReward
+            [],  # @note in case that whitelist feature would be leverage this will require review
+            [],  # @note in case that blacklist feature would be leverage this will require review
+            "Incentivizing eBTC ecosystem",
+        ],
+    )
+
+    # should help identifying on analytics $ebtc campaigns on the past
+    campaign_id = web3.solidityKeccak(["string"], ["EBTC_CAMPAIGN"]).hex()
+
+    campaign_params = (
+        campaign_id,
+        safe.address,
+        incentive_token.address,
+        token_amount_mantissa,
+        CAMPAIGN_TYPE_ERC20,  # campaignType can be: CLAMM: 2, ERC20: 1
+        start_date_ts,  # needs to be on the future
+        duration_campaign * SECONDS_PER_HOUR,  # duration in seconds
+        campaign_data,
+    )
+
+    # calculate fee
+    token_amount_mantissa_and_fee = _get_token_mantissa_and_fee(token_amount_mantissa)
+
+    # actions
+    incentive_token.approve(distribution_creator, token_amount_mantissa_and_fee)
+    distribution_creator.createCampaign(campaign_params)
+
+    safe.print_snapshot()
+
+    if not sim:
+        safe.post_safe_tx()
+
+
+# ===================== FEE AMOUNT CALCULATOR ===================== #
+# The following method is meant to be used for calculating the fee amount
+
+
+def _get_token_mantissa_and_fee(token_amount_mantissa):
+    base = distribution_creator.BASE_9()  # base is expressed on base 10 ** 9
+    fee = distribution_creator.defaultFees()
+    fee_mantissa = (token_amount_mantissa * fee) / base
+
+    return token_amount_mantissa + fee_mantissa
+
+
+# ===================== ARGUMENTS VERIFICATION HELPER ===================== #
+# The following method is meant to be used for checking the validity of the cli arguments
+
+
+def _verify_cli_args(
+    incentive_token,
+    token_amount,
+    weight_token_a,
+    weight_token_b,
+    weight_fees,
+    starting_date,
+    duration_campaign,
+):
+    valid_incentive_tokens = [
+        token_tuple[0] for token_tuple in distribution_creator.getValidRewardTokens()
+    ]
+    assert (
+        incentive_token.address in valid_incentive_tokens
+    ), "incentive token not valid!"
+
+    reward_token_min_amount = distribution_creator.rewardTokenMinAmounts(
+        incentive_token
+    )
+    token_amount_mantissa = int(token_amount) * 10 ** incentive_token.decimals()
+    incentive_token_balance = incentive_token.balanceOf(safe)
+    assert (
+        token_amount_mantissa > 0 and token_amount_mantissa <= incentive_token_balance,
+        "amount greater than balancer or zero",
```

Comment:
a typo: should be `balance` instead of `balancer`?

Reply:
you're right, fixed typo in 68f972f1f305cb372933af0a343c401fb14165b6